examples/fluids/src/misc.c

```diff
@@ -465,3 +466,136 @@ PetscErrorCode IntArrayP2C(PetscInt num_entries, PetscInt **array_petsc, CeedInt
+                        "    libCEED Backend                    : %s\n"
+                        "    libCEED Backend MemType            : %s\n",
+                        used_resource, CeedMemTypes[mem_type_backend]));
+  // PETSc
+  char box_faces_str[PETSC_MAX_PATH_LEN] = "3,3,3";
+  if (problem->dim == 2) box_faces_str[3] = '\0';
+  PetscCall(PetscOptionsGetString(NULL, NULL, "-dm_plex_box_faces", box_faces_str, sizeof(box_faces_str), NULL));
+  MatType mat_type;
+  VecType vec_type;
+  PetscCall(DMGetMatType(user->dm, &mat_type));
+  PetscCall(DMGetVecType(user->dm, &vec_type));
+  PetscCall(PetscPrintf(comm,
+                        "  PETSc:\n"
+                        "    Box Faces                          : %s\n"
+                        "    DM MatType                         : %s\n"
+                        "    DM VecType                         : %s\n"
+                        "    Time Stepping Scheme               : %s\n",
+                        box_faces_str, mat_type, vec_type, phys_ctx->implicit ? "implicit" : "explicit"));
+  if (user->app_ctx->cont_steps) {
+    PetscCall(PetscPrintf(comm,
+                          "  Continue:\n"
+                          "    Filename:                          : %s\n"
+                          "    Step:                              : %" PetscInt_FMT "\n"
+                          "    Time:                              : %g\n",
+                          user->app_ctx->cont_file, user->app_ctx->cont_steps, user->app_ctx->cont_time));
+  }
+  // Mesh
+  const PetscInt num_comp_q = 5;
+  PetscInt       glob_dofs, owned_dofs, local_dofs;
+  const CeedInt  num_P = user->app_ctx->degree + 1, num_Q = num_P + user->app_ctx->q_extra;
+  // -- Get global size
+  PetscCall(DMGetGlobalVectorInfo(user->dm, &owned_dofs, &glob_dofs, NULL));
+  // -- Get local size
+  PetscCall(DMGetLocalVectorInfo(user->dm, &local_dofs, NULL, NULL));
+  PetscCall(PetscPrintf(comm,
+                        "  Mesh:\n"
+                        "    Number of 1D Basis Nodes (P)       : %" CeedInt_FMT "\n"
+                        "    Number of 1D Quadrature Points (Q) : %" CeedInt_FMT "\n"
+                        "    Global DoFs                        : %" PetscInt_FMT "\n"
+                        "    DoFs per node                      : %" PetscInt_FMT "\n"
+                        "    Global nodes                       : %" PetscInt_FMT "\n",
+                        num_P, num_Q, glob_dofs, num_comp_q, glob_dofs / num_comp_q));
+  // -- Get Partition Statistics
+  PetscCall(PetscPrintf(comm, "  Partition:                             (min,max,median,max/median)\n"));
+  {
+    PetscInt    *gather_buffer = NULL;
+    PetscInt     part_owned_dofs[3], part_local_dofs[3], part_shared_dofs[3];
+    PetscInt     median_index = comm_size % 2 ? comm_size / 2 : comm_size / 2 - 1;
+    MPI_Datatype PetscInt_MPI;
+    PetscCall(PetscDataTypeToMPIDataType(PETSC_INT, &PetscInt_MPI));
```

Comment:
This works. You can also just use `MPIU_INT`, which is defined appropriately in `petscsys.h` (which is always included).